developer-docs-site/docs/move/book/functions.md

```diff
@@ -521,44 +521,61 @@ fun foo() { return () }
 
 ## Inline Functions
 
-Inline functions are functions which are expanded at caller side instead
-of compiled into Move bytecode. This allows to save gas and may lead
-to faster execution. For example, one can define an inline function
+Inline functions are functions whose bodies are expanded in place at the caller location during compile time.
+Thus, inline functions do not appear in Move bytecode as a separate functions: all calls to them are expanded away by the compiler.
+In certain circumstances, they may lead to faster execution and save gas.
+However, users should be aware that they could lead to larger bytecode size: excessive inlining potentially triggers various size restrictions.
+
+One can define an inline function by adding the `inline` keyword to a function declaration as shown below:
 
 ```move=
 inline fun percent(x: u64, y: u64):u64 { x * 100 / y }
 ```
 
-Now, when call `percent(2, 200)` the compiler will inline the function
-definition as if the user has written `2 * 100 / 200`.
+If we call this inline function as `percent(2, 200)`, the compiler will replace this call with the inline function's body, as if the user has written `2 * 100 / 200`.
+
+### Function parameters and lambda expressions
 
-### Function Parameters
+Inline functions support _function parameters_, which accept lambda expressions (i.e., higher-order functions) as arguments.
```

Comment:
Actually, lambda expressions are not necessarily "higher-order functions". 
 They are "anonymous functions".  Change this to "lambda expressions or anonymous functions as arguments".

Actually, I would be tempted to change the term to "anonymous function" everywhere and reference "lambda expressions" as a related term here.